tests/ios/tests/test_ios_course_home_tab.py

```diff
@@ -126,10 +127,41 @@ def test_validate_ui_elements(self, set_capabilities, setup_logging):
         component_header = global_contents.get_element_by_label_ios(set_capabilities, values.COURSE_COMPONENT_LABEL)
         assert component_header.text == values.COURSE_COMPONENT_LABEL
 
+    def test_component_navigation_smoke(self, set_capabilities, setup_logging):
+        """
+        Scenarios:
+            Verify next button element and it is clickable
+            Verify previous button element and it is clickable
+            Verify user can click on next button until finish button appears
+            Verify clicking finish button will load the celebratory modal
+            Verify Back to outline button on Modal and clicking this button
+                loads the compoenents screen
+            Verify clicking on back button will navigate the user to dashboard page
+        """
+
+        global_contents = Globals(setup_logging)
+        course_home_page = IosCourseHomeTab(set_capabilities, setup_logging)
+
+        next_btn = course_home_page.get_next_btn()
+        assert next_btn.text == values.COURSE_COMPONENT_NEXT_BUTTON
+        next_btn.click()
+
+        prev_btn = course_home_page.get_prev_btn()
+        assert prev_btn.text == values.COURSE_COMPONENT_PREVIOUS_BUTTON
+        prev_btn.click()
+
+        finish_button = course_home_page.component_navigation()
```

Comment:
what if we get exception in method?

Reply:
If we get an exception in this method, it will fail and return that exception by using the "NoSuchElementException" we have used inside this method.